### PR TITLE
Fixed #731 #730 and Add new ATR and fix memory leak issue

### PR DESCRIPTION
**Pull request by FeitianSmartcardReader**
The new pull request has fixed #730 and #731, and  add new ATR to pk01c card support, improve code and avoid memory leak issue.

### CODE REVIEW

**Comment by FeitianSmartcardReader** [comment]
@frankmorgner Have a check and let me know if everything OK, thanks

**Comment by FeitianSmartcardReader** [comment]
@frankmorgner can help merge branch to master? thanks

**Comment by frankmorgner** [comment]
@FeitianSmartcardReader This branch has conflicts that must be resolved.

File: src/libopensc/card-epass2003.c [hunk 1]
```diff
@@ -319,7 +319,7 @@ gen_init_key(struct sc_card *card, unsigned char *key_enc, unsigned char *key_ma
 	unsigned char iv[16] = { 0 };
 	
 	if (!card->drv_data) 
-		return SC_ERROR_INVALID_ARGUMENTS
+		return SC_ERROR_INVALID_ARGUMENTS;
 	
 	epass2003_exdata *exdata = (epass2003_exdata *)card->drv_data;
```
Comment (viktorTarasov) [inline src/libopensc/card-epass2003.c]: It's better to declare all variables at the beginning of block, before the first non-declaration instruction.

Comment (FeitianSmartcardReader) [inline src/libopensc/card-epass2003.c]: Good point, will do it soon, thanks @viktorTarasov

**Comment by frankmorgner** [comment]
Please reopen this PR when conflicts and the comments above are resolved.

**Pull request closed without merging**